File: workspace_java/BooleanExpression/src/br/unicamp/ic/bool/expression/BooleanOR.java
package br.unicamp.ic.bool.expression;

import br.unicamp.ic.bool.BooleanStore;

public class BooleanOR extends BooleanNode{
	private BooleanNode left, right;
	
	public BooleanOR(BooleanNode left, BooleanNode right)
	{
		super(bool_type.OR);
		this.left = left;
		this.right = right;
	}
	
	public BooleanNode getLeft()
	{
		return left;
	}
	
	public BooleanNode getRight()
	{
		return right;
	}
	
	public boolean compute(BooleanStore bs)
	{
		return (left.compute(bs) || right.compute(bs));
	}
	
	public String toString()
	{
		return "(" + left.toString() + " || " + right.toString() + ")";
	}
}

File: workspace_cpp/BooleanGraph/src/boolean/nodes/boolean_or.cpp
/*
 * BooleanOR.cpp
 *
 *  Created on: 5 de set de 2015
 *      Author: vroth
 */

#include "boolean_or.hpp"

BooleanOR::BooleanOR(BooleanNode * left, BooleanNode * right) : left(left), right(right) { btype = BOOLEAN_OR; }

BooleanOR::~BooleanOR() { }

std::string BooleanOR::toString() {
	return std::string("( " + left->toString() + " || " + right->toString() + ")");
}

bool BooleanOR::compute(BooleanStore *bs) {
	return left->compute(bs) || right->compute(bs);
}
